packages/splits-sdk-react/src/hooks/data.ts

```diff
@@ -104,6 +104,8 @@ export const useSplitEarnings = (
   )
   const [error, setError] = useState<RequestError>()
 
+  const stringErc20List =
```

Comment:
why do we have to stringify it again? sorry not to familiar with hooks

Reply:
so `useEffect` will re-run every time one of the variables in it's dependency array changes. For the `erc20TokenList`, since it's an array it depends on how it's passed in, but it might look like it's changing on each render.

Basically if you do this:
```
useSplitEarnings(1, '0x12345', true, ['0xusdc'])
```

then on every render, that array is going to be "new". meaning the useEffect is going to re-run every single time. You can fix this client side by doing something like this:
```
const erc20TokenList = useMemo(() => ['0xusdc'], [])
useSplitEarnings(1, '0x12345', true, erc20TokenList)
```

that way it's going to be the same array each time, meaning the useEffect won't get retriggered over and over. But that requires users of the react sdk to remember to define it like that, which they might not. So I want to handle it on our end. By converting it to a string, the `useEffect` will compare the previous string and the current string and it will see that it hasn't changed. The issue is when it compares objects or arrays, it's not comparing the values inside them but the actual data structure. Which again, could be "new" on every render depending on how it's defined by the client.

Happy to chat in person too if that's not making sense, not sure if I'm explaining it well

Reply:
ah, this makes sense, thank you!

Reply:
v sad react hasn't figured out a better solution here yet,,